### PR TITLE
Add polyfills and Stalker stubs for Duktape

### PR DESCRIPTION
**Pull request by s1341**

### CODE REVIEW

File: bindings/gumjs/harmony-collections.js [hunk 1]
```diff
@@ -0,0 +1,795 @@
+/* (The MIT License)
```
Comment (oleavr) [inline bindings/gumjs/harmony-collections.js]: Would be good to rename this file to `gumjs-collections.js` for consistency.

File: bindings/gumjs/Makefile.am [hunk 2]
```diff
@@ -137,6 +137,10 @@ duk_sources = \
 	gumduksymbol.c \
 	gumdukinstruction.h \
 	gumdukinstruction.c \
+	gumdukpolyfill.h \
+	gumdukpolyfill.c \
```
Comment (oleavr) [inline bindings/gumjs/Makefile.am]: Let's do this in js instead, it's just 3 lines of code or so.

File: bindings/gumjs/gumdukobject.c [hunk 3]
```diff
@@ -58,16 +58,12 @@ _gumjs_duk_create_subclass (duk_context * ctx,
 }
 
 void
-_gumjs_duk_add_properties_to_class (duk_context * ctx,
-                                    const gchar * class_name,
-                                    const GumDukPropertyEntry * entries)
+_gumjs_duk_add_properties_to_class_by_heapptr (duk_context * ctx, GumDukHeapPtr klass,
+    const GumDukPropertyEntry * entries)
```
Comment (oleavr) [inline bindings/gumjs/gumdukobject.c]: Reverting coding style fixes?

File: bindings/gumjs/gumdukobject.c [hunk 4]
```diff
@@ -76,26 +72,40 @@ _gumjs_duk_add_properties_to_class (duk_context * ctx,
...
+    // [ proto ]
   }
 
+  duk_pop (ctx);
+  // []
+}
+
+void
+_gumjs_duk_add_properties_to_class (duk_context * ctx, gchar * classname,
+    const GumDukPropertyEntry * entries)
```
Comment (oleavr) [inline bindings/gumjs/gumdukobject.c]: Reverting coding style fixes?

File: bindings/gumjs/gumdukobject.c [hunk 5]
```diff
@@ -76,26 +72,40 @@ _gumjs_duk_add_properties_to_class (duk_context * ctx,
...
+{
+  const GumDukPropertyEntry * entry;
+  duk_get_global_string (ctx, classname);
+  // [ class ]
+  duk_get_prop_string (ctx, -1, "prototype");
+  // [ class proto ]
+  _gumjs_duk_add_properties_to_class_by_heapptr (ctx, duk_require_heapptr (ctx, -1), entries);
```
Comment (oleavr) [inline bindings/gumjs/gumdukobject.c]: > 80 cols.

File: bindings/gumjs/gumdukobject.c [hunk 6]
```diff
@@ -76,26 +72,40 @@ _gumjs_duk_add_properties_to_class (duk_context * ctx,
...
 
+  duk_pop (ctx);
+  // []
+}
+
+void
+_gumjs_duk_add_properties_to_class (duk_context * ctx, gchar * classname,
+    const GumDukPropertyEntry * entries)
+{
+  const GumDukPropertyEntry * entry;
```
Comment (oleavr) [inline bindings/gumjs/gumdukobject.c]: Missing blank line after variable declaration(s).

File: bindings/gumjs/gumdukobject.h [hunk 7]
```diff
@@ -21,13 +21,16 @@ struct _GumDukPropertyEntry
...
-    const gchar * class_name);
+G_GNUC_INTERNAL void _gumjs_duk_create_subclass (duk_context * ctx, const gchar * parent, const gchar * name,
+    gpointer constructor, gint constructor_nargs, gpointer finalize);
+
+G_GNUC_INTERNAL void _gumjs_duk_add_properties_to_class_by_heapptr (duk_context * ctx, GumDukHeapPtr klass,
+    const GumDukPropertyEntry * entries);
+G_GNUC_INTERNAL void _gumjs_duk_add_properties_to_class (duk_context * ctx, gchar * classname,
+    const GumDukPropertyEntry * entries);
+
+G_GNUC_INTERNAL gboolean _gumjs_is_arg0_equal_to_prototype (duk_context * ctx, const gchar * classname);
```
Comment (oleavr) [inline bindings/gumjs/gumdukobject.h]: Reverting back to > 80 cols?

File: bindings/gumjs/gumdukstalker.c [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+  self->stalker = NULL;
+  self->queue_capacity = 16384;
+  self->queue_drain_interval = 250;
+
+  duk_push_c_function (ctx, gumjs_stalker_construct, 0);
+  // [ construct ]
+  duk_push_object (ctx);
+  // [ construct proto ]
+  duk_put_function_list (ctx, -1, gumjs_stalker_functions);
+  _gumjs_duk_add_properties_to_class_by_heapptr (ctx, duk_require_heapptr (ctx, -1), gumjs_stalker_values);
```
Comment (oleavr) [inline bindings/gumjs/gumdukstalker.c]: Exceeding 80 cols.

File: bindings/gumjs/gumdukstalker.c [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+GUMJS_DEFINE_CONSTRUCTOR (gumjs_stalker_construct)
+{
+  return 0;
+}
+
+GUMJS_DEFINE_GETTER (gumjs_stalker_get_trust_threshold)
+{
+  GumStalker * stalker;
+
+  stalker = _gum_duk_stalker_get (_gumjs_get_private_data (ctx, _gumjs_duk_get_this (ctx)));
```
Comment (oleavr) [inline bindings/gumjs/gumdukstalker.c]: Exceeding 80 cols.

File: bindings/gumjs/gumdukstalker.c [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+  duk_push_number (ctx, gum_stalker_get_trust_threshold (stalker));
+  return 1;
+}
+
+GUMJS_DEFINE_SETTER (gumjs_stalker_set_trust_threshold)
+{
+  GumStalker * stalker;
+  gint threshold;
+
+  stalker = _gum_duk_stalker_get (_gumjs_get_private_data (ctx, _gumjs_duk_get_this (ctx)));
```
Comment (oleavr) [inline bindings/gumjs/gumdukstalker.c]: Exceeding 80 cols.

File: bindings/gumjs/gumdukobject.c (hunk 3 above)
Comment (s1341) [inline bindings/gumjs/gumdukobject.c]: no. I added a function. note the _by_heapptr suffix

File: bindings/gumjs/gumdukobject.c (hunk 4 above)
Comment (s1341) [inline bindings/gumjs/gumdukobject.c]: Added a function, as above.

Comment (oleavr) [inline bindings/gumjs/gumdukobject.c]: Ok. Note that the function declaration style is different between .h and .c, and here we're using the wrong one.

File: bindings/gumjs/gumdukobject.c (hunk 3 above)
Comment (oleavr) [inline bindings/gumjs/gumdukobject.c]: Ok. Style is off as arguments should be aligned (see other comment).

**Pull request merged**